input:
  pointcloud:
    subscribe_to_curbs: false
    topic: /habitat/points
    fields: xyzrgb
    rotation_matrix: [[1, 0, 0], [0, 1, 0], [0, 0, 1]]
    floor_height: -0.9
    ceiling_height: 1.5
  subscribe_to_gt_pose: true
  gt_pose:
    topic: /true_pose
    type: PoseStamped
  odometry:
    topic: /odom_noised
  subscribe_to_images: true
  image_front:
    topic: /habitat/rgb1/image
    color: rgb
  image_back:
    topic: /habitat/rgb3/image
    color: rgb
topomap:
  mode: localization
  #start_location: 0
  #start_local_pose: [0, 0, 0]
  iou_threshold: 0.3
  max_edge_length: 8.0
  drift_coef: 0.2
  localization_frequency: 3.0
  localization_timeout: 30.0
  rel_pose_correction_frequency: 1.0
place_recognition:
  model: minkloc3d
  pointcloud_quantization_size: 0.2
  # Path to weights and model config relative to OpenPlaceRecognition directory
  weights_path: weights/place_recognition/minkloc3d_nclt.pth
  model_config_path: configs/model/place_recognition/resnet18-seqgem_minkloc3dv2.yaml
  top_k: 5
local_occupancy_grid:
  resolution: 0.1
  radius: 18.0
  max_range: 15.0
scan_matching:
  model: feature2d
  detector_type: ORB
  score_threshold: 0.4
  outlier_thresholds: [5, 2, 1, 0.5, 0.5]
  voxel_downsample_size: 0.1
  min_matches: 5
scan_matching_along_edge:
  model: feature2d
  detector_type: HarrisWithDistance
  score_threshold: 0.4
  outlier_thresholds: [5, 2, 1, 0.5, 0.5]
  voxel_downsample_size: 0.1
  min_matches: 5
  jump_threshold: 0.5
visualization:
  publish_gt_map: true
  map_frame: map
  publish_tf_from_odom: false